src/framework/ui/card/card.component.tsx

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import React from 'react';
+import {
+  StyleSheet,
+  View,
+  StyleProp,
+  TextStyle,
+  ViewStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  GestureResponderEvent,
+} from 'react-native';
+import {
+  styled,
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import {
+  Divider,
+  DividerElement,
+} from '../divider/divider.component';
+import { CardHeaderElement } from './cardHeader.component';
+import { allWithPrefix } from '../support/services';
+
+interface HeaderStyles {
+  style: StyleProp<ViewStyle>;
+  accent: StyleProp<ViewStyle>;
+  title: StyleProp<TextStyle>;
+  description: StyleProp<TextStyle>;
+}
+
+type HeaderProp = React.ReactElement<any> | CardHeaderElement;
+type FooterProp = React.ReactElement<any>;
+export type CardFooterElement = FooterProp;
+
+interface ComponentProps {
+  appearance?: string;
+  status?: string;
+  children: React.ReactNode;
+  header?: () => HeaderProp;
+  footer?: () => FooterProp;
+}
+
+export type CardProps = StyledComponentProps & TouchableOpacityProps & ComponentProps;
+export type CardElement = React.ReactElement<CardProps>;
+
+/**
+ * Styled `Card` component is a basic content container component.
+ *
+ * @extends React.Component
+ *
+ * @property {string} status - Determines the status of the component.
+ * Can be `basic`, `primary`, `success`, `info`, `warning`, `danger`.
+ * Default is `basic`.
+ *
+ * @property {string} appearance - Determines the appearance of the component.
+ * Can be `filled` or `outline`.
+ * Default is `outline`.
+ *
+ * @property {React.ReactNode} children - Determines text of the component.
+ *
+ * @property {() => HeaderProp} header - Determines header of the component.
+ *
+ * @property {() => FooterProp} footer - Determines footer of the component.
+ *
+ * @property {TouchableOpacityProps} - Any props applied to TouchableOpacity component.
+ *
+ * @property {StyledComponentProps} - Any props applied to `styled` component.
+ *
+ * @overview-example CardSimpleUsage
+ *
+ * @overview-example CardStatuses
+ *
+ * @overview-example CardEvaHeader
+ *
+ * @overview-example CardCustomHeader
+ *
+ * @overview-example CardFooter
+ *
+ * @overview-example CardHeaderFooter
+ */
+
+class CardComponent extends React.Component<CardProps> {
+
+  static styledComponentName: string = 'Card';
+
+  private onPress = (event: GestureResponderEvent): void => {
```

Comment:
no need to override `onPress`, `onPressIn` and `onPressOut` events since nothing additional is called here. This will make it work when you just do `{...restProps}`